paddle/majel/allocation.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+#include <memory>
```

Comment:
Is `<memory>` mandatory in this header file? Should we move it to `allocation.cc`?